### ISSUE TITLE
A question regarding cross-entropy loss

### ISSUE THREAD

**Issue by sunsunnyshine** [TonyLianLong/RCF-UnsupVideoSeg#4]
I observed a possible bug in the cross-entropy calculation in the code. However, even after making modifications, I still noticed oscillations in the loss resembling a sine wave, and these oscillations are present in all loss values. Have you also observed this phenomenon? What could be the reason behind it?
The modified code section is as follows：
https://github.com/TonyLianLong/RCF-UnsupVideoSeg/blob/1ec74f307dc7149b5a656bad7375222671dd9efc/models/rcf_model.py#L433C37-L433C37
I modified to：
` all_pred_mask_1 = all_pred_mask.view(batch_size, im_num, self.mask_layer, _feat_h, _feat_w)`
` all_pred_mask = F.softmax(all_pred_mask_1, dim=2)`
` log_all_pred_mask = F.log_softmax(all_pred_mask_1, dim=2)`
and the observed loss curve is shown below:
![image](https://github.com/TonyLianLong/RCF-UnsupVideoSeg/assets/87694659/8d065086-b66b-4254-ab38-e8b9525eeb82)

**Comment by sunsunnyshine** [comment]
I found the issue; a bug in the log recording was causing oscillations in the visualization of the loss. I made the following modifications to the code, but incorporating the corrected cross-entropy doesn't seem to improve performance.
https://github.com/TonyLianLong/RCF-UnsupVideoSeg/blob/1ec74f307dc7149b5a656bad7375222671dd9efc/main.py#L171C47-L171C47
修改如下：
![image](https://github.com/TonyLianLong/RCF-UnsupVideoSeg/assets/87694659/897166b7-690c-42d2-882f-36130e9357c7)
